src/openms/source/DATASTRUCTURES/QTCluster.cpp

```diff
@@ -179,6 +180,7 @@ namespace OpenMS
     // Store best (closest) element:
     // Only add the element if either no element is present for the map or if
     // the element is closer than the current element for that map
+    //TODO shouldn't IDed features be preferred to unIDed ones?
```

Comment:
yep e.g., if it would be possible to first cluster ided features (e.g., by setting distance to infinity) and then unided this could help potentially

Reply:
Yes I started on that as well. It is possible but a bit complicated since most of the code is not written in a reusable way to do incremental clustering.
But what I thought here: Since it is a normalized distance (0-1) we could add 1 to unidentified features so IDed ones are always preferred but not used exclusively.

Reply:
hmm sounds like it could work